hooks/tk-multi-workfiles2/scene_operation_tk-maya.py

```diff
@@ -0,0 +1,236 @@
+        """
+        Main hook entry point
+
+        :param operation:       String
+                                Scene operation to perform
+
+        :param file_path:       String
+                                File path to use if the operation
+                                requires it (e.g. open)
+
+        :param context:         Context
+                                The context the file operation is being
+                                performed in.
+
+        :param parent_action:   This is the action that this scene operation is
+                                being executed for.  This can be one of:
+                                - open_file
+                                - new_file
+                                - save_file_as
+                                - version_up
+
+        :param file_version:    The version/revision of the file to be opened.  If this is 'None'
+                                then the latest version should be opened.
+
+        :param read_only:       Specifies if the file should be opened read-only or not
+
+        :returns:               Depends on operation:
+                                'current_path' - Return the current scene
+                                                 file path as a String
+                                'reset'        - True if scene was reset to an empty
+                                                 state, otherwise False
+                                all others     - None
+        """
+
+        if operation == "current_path":
+            # return the current scene path
+            return cmds.file(query=True, sceneName=True)
+        elif operation == "open":
+            # do new scene as Maya doesn't like opening
+            # the scene it currently has open!
+            cmds.file(new=True, force=True)
+            cmds.file(file_path, open=True, force=True)
+
+            # getting fields from file path because context.as_template_fields() doesn't contain {name}, {version}
+            file_template = context.sgtk.template_from_path(file_path)
+            fields = file_template.get_fields(file_path)
+
+            render_temp = self.get_render_template(context)
+            frame_sq_key = context.sgtk.template_keys['SEQ']    # Can 'SEQ' change?
+
+            # get show resolution
+            indiapipeline_prefs = preferences.Preferences(package="indiapipeline")
+            try:
+                fields.update({"width": indiapipeline_prefs["show_resolution"]["width"],
+                               "height": indiapipeline_prefs["show_resolution"]["height"],
+                               "output": "LAYERPLACEHOLDER"})   # output is alphanumeric; replace with token <Layer>
+            except KeyError as ke:
+                # logging?
+                print "Unable to find {} in indiapipeline preferences. Not setting render defaults.".format(ke)
```

Comment:
Please use the SGTK logging module: self.parent.log_[info|warning|error|debug], or the newer self.parent.logger.[info|warning|error|debug]

Reply:
Ah, forgot to do that.